service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -24,16 +26,35 @@ public ConfigServiceImpl(ConfigRepository configRepository) {
 
     @Override
     public Optional<ClientTestProperty> getClientTestProperty(final String clientName, final String assessmentId) {
-        ClientTestProperty clientTestProperty = configRepository.getClientTestProperty(clientName, assessmentId);
-
-        return Optional.ofNullable(clientTestProperty);
+        return configRepository.getClientTestProperty(clientName, assessmentId);
     }
 
     @Override
     public Optional<ClientSystemFlag> getClientSystemFlag(String clientName, String auditObject) {
         List<ClientSystemFlag> clientSystemFlags = configRepository.getClientSystemFlags(clientName);
 
-        return clientSystemFlags.stream().filter(f -> f.getAuditObject().equals(auditObject))
+        return clientSystemFlags.stream()
+                .filter(f -> f.getAuditObject().equals(auditObject))
                 .findFirst();
     }
+
+    @Override
+    public Optional<Integer> getTaCheckInTimeLimit(final String clientName) {
+        Optional<TimeLimits> timeLimits =  configRepository.getTimeLimits(clientName);
+
+        if (timeLimits.isPresent()) {
+            return Optional.of(
+                    timeLimits.map(TimeLimits::getTaCheckinTimeMinutes)
+                    .orElse(0));
```

Comment:
If we are in this conditional, then timeLimits has a value, and since getTaCheckinTimeMinutes returns an Integer and not an Optional<Integer>, can the .orElse(0) ever get called?  I feel like I'm missing something here, so figured I'd check.